src/main/java/hudson/tasks/junit/ClassResult.java

```diff
@@ -189,6 +200,21 @@ else if(r.isPassed()) {
                 failCount++;
             }
             duration += r.getDuration();
+            if (arId == null) {
```

Comment:
so this looks a little funky to me...
the results are including all cases regardless of the archiveId that the result was from.